packages/nc-gui/components/nc/Pagination.vue

```diff
@@ -11,19 +12,41 @@ const props = defineProps<{
   nextPageTooltip?: string
   firstPageTooltip?: string
   lastPageTooltip?: string
+  showSizeChanger?: boolean
 }>()
 
 const emits = defineEmits(['update:current', 'update:pageSize'])
 
-const { total } = toRefs(props)
+const { total, showSizeChanger } = toRefs(props)
 
 const current = useVModel(props, 'current', emits)
 
 const pageSize = useVModel(props, 'pageSize', emits)
 
+const localPageSize = computed({
+  get: () => {
+    if (!showSizeChanger.value) return pageSize.value
+
+    const storedPageSize = !isNaN(parseInt(localStorage.getItem(NC_GRID_VIEW_PAGE_SIZE_KEY) ?? ''))
+      ? parseInt(localStorage.getItem(NC_GRID_VIEW_PAGE_SIZE_KEY) ?? '')
+      : 25
+
+    if (pageSize.value !== storedPageSize) {
+      pageSize.value = storedPageSize
+    }
+
+    return pageSize.value
+  },
+  set: (val) => {
+    localStorage.setItem(NC_GRID_VIEW_PAGE_SIZE_KEY, `${val}`)
+
+    pageSize.value = val
+  },
```

Comment:
For persisting value in localstorage you can use `useGlobal` composable - `packages/nc-gui/composables/useGlobal/state.ts`

Reply:
fixed in pr: #8300